### ISSUE TITLE
Memory Consumption | Boosting model

### ISSUE THREAD

**Issue by samanemami** [tensorflow/tensorflow#52067]
<em> This is an issue related to performance of [TensorFlow.BoostedTree](https://www.tensorflow.org/tutorials/estimator/boosted_trees)</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Mint
- TensorFlow installed from (source or binary): pip
- TensorFlow version (use command below): '2.4.1'
- Python version:  3.6.5

**Describe the model**

I am training the Boosted Tree of the TensorFlow to review the memory consumption of the model by using the tracemalloc.
This model is based on the Gradient Boosting and XGBoost idea.

**Describe the current behavior**

The trend of the model is not growing linearly and has too much noise in terms of memory consumption.

**Describe the expected behavior**

As the XGboost demonstrates, it should grow linearly.

**Possible cause**
I think there is something inside tf.estimator that causes this behavior, if it is so, would you please indicate the exact reason?

**Standalone code to reproduce the issue**
I provided a notebook over a UCI Dataset in this [link](https://github.com/samanemami/TFBoostedTree/blob/main/examples/memory_consumption_tensorflow.ipynb).
Which shows the memory consumption of the model.
